src/mechanics/input.rs

```diff
@@ -1,628 +1,594 @@
-            PLAYER_MIDPOINT as f32,
-            0.0,
-        );
-        let actual_transform =
-            *player_query.expect("out_of_bounds_bottomright [test]: Player could not be found");
-
-        assert_eq!(expected_transform, actual_transform);
+pub fn move_player_on_key_press(
+    input: Res<ButtonInput<KeyCode>>,
+    mut move_player_requester: EventWriter<MovementDirection>,
+) {
+    if input.pressed(KeyCode::KeyW) {
+        move_player_requester.send(MovementDirection::Up);
+    } else if input.pressed(KeyCode::KeyS) {
+        move_player_requester.send(MovementDirection::Down);
+    } else if input.pressed(KeyCode::KeyA) {
+        move_player_requester.send(MovementDirection::Left);
+    } else if input.pressed(KeyCode::KeyD) {
+        move_player_requester.send(MovementDirection::Right);
     }
+}
 
-    #[test]
-    fn out_of_bounds_top() {
-        let mut app = setup_app_bounds_checking();
-
-        let player_id = app
-            .world
-            .spawn_empty()
-            .insert((
-                Player,
-                Transform::from_xyz(
-                    TEST_LEVEL_WIDTH_IN_BOUNDS,
-                    TEST_LEVEL_HEIGHT_OUT_TBOUNDS,
-                    0.0,
-                ),
-            ))
-            .id();
-
-        app.update();
-
-        let player_query = app.world.get::<Transform>(player_id);
-        assert!(player_query.is_some());
-
-        let expected_transform = Transform::from_xyz(
-            TEST_LEVEL_WIDTH_IN_BOUNDS,
-            (TEST_LEVEL_HEIGHT - PLAYER_MIDPOINT) as f32,
-            0.0,
-        );
-        let actual_transform =
-            *player_query.expect("out_of_bounds_top [test]: Player could not be found");
+//pub fn animate_entity(
+//    mut entity_query: Query<(&mut TextureAtlas, &DirectionFacing), Changed<DirectionFacing>>,
+//) {
+//    if entity_query.is_empty() {
+//        return;
+//    }
+//
+//    for (mut sprite, facing) in entity_query.iter_mut() {
+//        match facing {
+//            DirectionFacing::Up => {
+//                sprite.index = 0;
+//            }
+//            DirectionFacing::Down => {
+//                sprite.index = 1;
+//            }
+//            DirectionFacing::Left => {
+//                sprite.index = 2;
+//            }
+//            DirectionFacing::Right => {
+//                sprite.index = 3;
+//            }
+//        }
+//    }
+//}
+
+//pub fn move_entity(
+//    mut entity_query: Query<
+//        (&mut Transform, &DirectionFacing, &mut MovementIntent),
+//        Changed<MovementIntent>,
+//    >,
+//    tile_query: Query<&EntityInstance>,
+//    level_dimension: Res<LevelDimensions>,
+//    mut entity_movement_broadcast: EventWriter<PlayerMovementActions>,
+//) {
+//    if entity_query.is_empty() {
+//        return;
+//    }
+//
+//    let collision_tiles = tile_query
+//        .iter()
+//        .filter(|&tile| !tile.field_instances.is_empty())
+//        .filter(|&tile| {
```

Comment:
We should be able to remove this commented out function now, unless we have this here as reference for something else in the future.

Reply:
Looks like we've replaced all parts of this function, it should be fine to remove it.